File: backend/app/constants/avatar_constants.py
"""
Constantes relacionadas con el sistema de avatares de usuarios.
"""

from typing import List

# Tipos de avatar permitidos
AVATAR_TYPES = {
    "initials": "initials",  # Avatar generado con iniciales
    "hombre": "hombre",      # Avatar predefinido masculino
    "mujer": "mujer"         # Avatar predefinido femenino
}

# Rutas de avatares predefinidos
AVATAR_PREDEFINED_PATHS = {
    "hombre": "/img/profile/male-default.svg",
    "mujer": "/img/profile/female-default.svg"
}

# Configuración de archivos de imagen
MAX_AVATAR_SIZE_BYTES = 5 * 1024 * 1024  # 5MB
ALLOWED_IMAGE_TYPES = ["image/jpeg", "image/png", "image/jpg", "image/webp"]
ALLOWED_IMAGE_EXTENSIONS = [".jpg", ".jpeg", ".png", ".webp"]

# Directorio de almacenamiento
AVATAR_UPLOAD_DIR = "uploads/profiles"

File: frontend/constants/avatarConstants.js
/**
 * Constantes relacionadas con el sistema de avatares de usuarios.
 */

// Tipos de avatar permitidos
export const AVATAR_TYPES = {
  CUSTOM: 'custom',
  INITIALS: 'initials',
  HOMBRE: 'hombre',
  MUJER: 'mujer'
};

// Rutas de avatares predefinidos
export const AVATAR_PATHS = {
  DEFAULT_HOMBRE: '/avatar-male-default.png',
  DEFAULT_MUJER: '/avatar-female-default.png'
};

// Mapeo de tipos a rutas
export const AVATAR_TYPE_MAP = {
  [AVATAR_TYPES.HOMBRE]: AVATAR_PATHS.DEFAULT_HOMBRE,
  [AVATAR_TYPES.MUJER]: AVATAR_PATHS.DEFAULT_MUJER
};

// Configuración de archivos
export const MAX_AVATAR_SIZE_MB = 5;
export const MAX_AVATAR_SIZE_BYTES = MAX_AVATAR_SIZE_MB * 1024 * 1024;
export const ALLOWED_IMAGE_TYPES = ['image/jpeg', 'image/png', 'image/jpg', 'image/webp'];

// Configuración de SVG para iniciales
export const INITIALS_SVG_CONFIG = {
  width: 200,
  height: 200,
  radius: 100,
  backgroundColor: '#1B5E9F',
  textColor: 'white',
  fontSize: 80,
  fontFamily: 'Arial, sans-serif',
  fontWeight: 'bold'
};
